src/Blockcore.Indexer.Cirrus/Storage/Mongo/SmartContracts/DaoContractAggregator.cs

```diff
@@ -0,0 +1,138 @@
+   }
+
+   async Task<DaoContractComputedTable> LookupDaoContractForAddressAsync(string address)
+   {
+      DaoContractComputedTable contract = await db.DaoContractComputedTable
+         .AsQueryable()
+         .SingleOrDefaultAsync(_ => _.ContractAddress == address);
+
+      if (contract is not null)
+         return contract;
+
+      var contractCode = await db.CirrusContractCodeTable
+         .AsQueryable()
+         .SingleOrDefaultAsync(_ => _.ContractAddress == address);
+
+      if (contractCode is null || contractCode.CodeType != DaoContract)
+      {
+         logger.LogInformation(
+            $"Request to compute DAO contract for address {address} which was not found in the contract code table");
+         return null;
+      }
+
+      contract = await CreateNewDaoContract(address);
+
+      if (contract is null)
+         throw new ArgumentNullException($"Contract not found in the contract table for address {address}");
+
+      return contract;
+   }
+
+   private async Task<DaoContractComputedTable> CreateNewDaoContract(string address)
+   {
+      var contractCreationTransaction = await db.CirrusContractTable
+         .AsQueryable()
+         .Where(_ => _.NewContractAddress == address)
+         .SingleOrDefaultAsync();
+
+      if (contractCreationTransaction is null)
+         throw new ArgumentNullException(nameof(contractCreationTransaction));
+
+      var contract = new DaoContractComputedTable
+      {
+         ContractAddress = contractCreationTransaction.NewContractAddress,
+         ContractCreateTransactionId = contractCreationTransaction.TransactionId,
+         LastProcessedBlockHeight = contractCreationTransaction.BlockIndex
+      };
+
+      await SaveTheContractAsync(address, contract);
+
+      return contract;
+   }
+
+   private async Task AddNewTransactionsDataToDocumentAsync(string address, List<CirrusContractTable> contractTransactions,
+      DaoContractComputedTable contract)
+   {
+      foreach (var contractTransaction in contractTransactions)
+      {
+         var reader = logReaderFactory.GetLogReader(contractTransaction.MethodName);
+
+         if (reader is null)
+         {
+            Console.WriteLine(contractTransaction.MethodName);
+            continue; //TODO need to verify this is the right way to go
+         }
+
+         if (!reader.IsTheTransactionLogComplete(contractTransaction.Logs))
+         {
+            var result = await cirrusClient.GetContractInfoAsync(contractTransaction.TransactionId);
+
+            contractTransaction.Logs = result.Logs;
+         }
+
+         reader.UpdateContractFromTransactionLog(contractTransaction, contract);
+
+         contract.LastProcessedBlockHeight = contractTransaction.BlockIndex;
+      }
+
+      await SaveTheContractAsync(address, contract);
+   }
+
+   async Task SaveTheContractAsync(string address, DaoContractComputedTable contract) =>
+      await db.DaoContractComputedTable.FindOneAndReplaceAsync<DaoContractComputedTable>(
```

Comment:
I think we should add logic here for the modification condition where if the block height in store right now is not the same as it was when we fetched the document we would fail the update.
see the logic in the computed calculation for utxo history 
https://github.com/block-core/blockcore-indexer/blob/master/src/Blockcore.Indexer.Core/Storage/Mongo/MongoData.cs#L873